components/textfield/index.css

```diff
@@ -131,207 +458,233 @@ governing permissions and limitations under the License.
+    }
   }
 
-  .spectrum-Textfield.is-valid & {
-    padding-inline-end: calc(
-      var(--spectrum-textfield-texticon-padding-right) +
-        var(--spectrum-textfield-texticon-success-icon-width) +
-        var(--spectrum-textfield-icon-inline-end-override, var(--spectrum-textfield-texticon-success-icon-margin-left))
-    );
+  /*** Input Invalid ⚠️ ***/
+  .is-invalid & {
+    border-color: var(--highcontrast-textfield-border-color-invalid-default, var(--mod-textfield-border-color-invalid-default, var(--spectrum-textfield-border-color-invalid-default)));
   }
 
-  .spectrum-Textfield.is-invalid & {
-    padding-inline-end: calc(
-      var(--spectrum-textfield-texticon-padding-right) +
-        var(--spectrum-textfield-texticon-invalid-icon-width) +
-        var(--spectrum-textfield-icon-inline-end-override, var(--spectrum-textfield-texticon-invalid-icon-margin-left))
-    );
+  /* Invalid hover */
+  .is-invalid:hover &,
+  .is-invalid &:hover {
+    border-color: var(--highcontrast-textfield-border-color-invalid-hover, var(--mod-textfield-border-color-invalid-hover, var(--spectrum-textfield-border-color-invalid-hover)));
   }
 
-  .spectrum-Textfield--multiline & {
-    block-size: var(--spectrum-textarea-height-adjusted);
-    min-block-size: var(--spectrum-textarea-min-height);
-    padding: var(--spectrum-textarea-padding-top)
-      var(--spectrum-textarea-padding-right)
-      var(--spectrum-textarea-padding-bottom)
-      calc(var(--spectrum-textarea-padding-left) - 1px);
+  /* Invalid mouse focus */
+  .is-invalid.is-focused &,
+  .is-invalid:focus &,
+  .is-invalid &:focus {
+    border-color: var(--highcontrast-textfield-border-color-invalid-focus, var(--mod-textfield-border-color-invalid-focus, var(--spectrum-textfield-border-color-invalid-focus)));
 
-    /* Remove the default vertical scrollbar for textarea in IE. */
-    overflow: auto;
+    /* focus hover */
+    &:hover {
+      border-color: var(--highcontrast-textfield-border-color-invalid-focus-hover, var(--mod-textfield-border-color-invalid-focus-hover, var(--spectrum-textfield-border-color-invalid-focus-hover)));
+    }
   }
 
-  .spectrum-Textfield--quiet & {
-    border-radius: var(--spectrum-textfield-quiet-texticon-border-radius);
-    border-block-start-width: 0;
-    border-block-end-width: var(--spectrum-textfield-quiet-texticon-border-bottom-size);
-    border-inline-width: 0;
+  /* invalid keyboard focus */
+  .is-invalid.is-keyboardFocused &,
+  &:focus-ring,
+  &:focus-visible {
+    border-color: var(--highcontrast-textfield-border-color-invalid-keyboard-focus, var(--mod-textfield-border-color-invalid-keyboard-focus, var(--spectrum-textfield-border-color-invalid-keyboard-focus)));
+  }
+
+  /****** Input Disabled 🚫 ******/
+  .spectrum-Textfield.is-disabled &,
+  .spectrum-Textfield.is-disabled:hover &,
+  &:disabled {
+    background-color: var(--mod-textfield-background-color-disabled, var(--spectrum-textfield-background-color-disabled));
```

Comment:
```suggestion
    background-color: var(--highcontrast-textfield-background-color-disabled, var(--mod-textfield-background-color-disabled, var(--spectrum-textfield-background-color-disabled)));
```

Reply:
In WHCM, the default behavior of the background color on the ddisabled component as it is on the current docs site is white (black for dark mode).
<img width="1099" alt="Screen Shot 2022-11-21 at 2 54 07 PM" src="https://user-images.githubusercontent.com/25614178/203146522-25238763-ba24-4236-ba4e-f79b8977d6a6.png">

Do you think we should be overriding it to something else? @yosevu

Reply:
I see. I'm not sure, but why not use an explicit `ButtonFace` anyway to represent this in WHCM? I think it is more consistent and would allow the background color to be controlled in the same way as the other system colors.